src/os/envi_wasi.go

```diff
@@ -0,0 +1,78 @@
+// +build wasi
+
+package os
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+// TODO: this is not thread safe. When we support threading proposal, we need to reconsider the implementation.
```

Comment:
if we invoke `ensureEnv` in the init function, then we do not need to care about this even if we support threading proposal. But I'm not sure we should always call `environ_get` and environ_sizes_get`.